### ISSUE TITLE
[Helm] Support deploying a Fleet Server Elastic Agent

### ISSUE THREAD

**Issue by eedugon** [elastic/fleet-server#4233]
We should support the installation of a Fleet Server on Kubernetes with the helm chart.
Deploying a Fleet Server agent is slightly different than deploying a normal Fleet Managed Elastic Agent.

It requires:
- `FLEET_SERVER_ENABLE` set to true.
- A policy ID of the policy prepared in Kibana including Fleet Server integration, for `FLEET_SERVER_POLICY_ID` env var.
- A Service Token (and not an enrollment token), for `FLEET_SERVER_SERVICE_TOKEN`
- Elasticsearch URL, for `FLEET_SERVER_ELASTICSEARCH_HOST `
- The CA certificate of Elasticsearch, as a file pointed by `ELASTICSEARCH_CA` environment variable, or the **SHA fingerprint of the CA** through `FLEET_SERVER_ELASTICSEARCH_CA_TRUSTED_FINGERPRINT` env var (both options are in theory possible).
- A certificate and key for the HTTP endpoint (if not provided the Fleet Server generates a self-signed cert, which is called `quick start mode` in the documentation, but we should clearly support providing the cert and key as it's the requirement for production).

^^ From the previous items we don't expect the helm chart to take care of everything, probably the Policy ID and Service Token should be prepared in advance by the user and provided through `values.yaml`. Same for the Elasticsearch URL and the CA cert of Elasticsearch.

This type of Agent would NOT require any interaction with Kubernetes API, so I would suggest to NOT add any RBAC resources (probably neither a serviceAccount) --> `automountServiceAccountToken: false` should be the way to go here.

We have recently added a doc about installing Fleet Server on Kubernetes (without the chart of course :) ) that could be used as a reference of the needed resources (a Deployment + 2 secrets would be enough).

About the certificate + key we need to decide how to implement it, as there are multiple options:
- Request the user to provide the cert + key for the fleet server in a separate secret (this is probably the easiest).
- Offer some kind of automation or integration with something like cert manager.

Note that the previous item about cert + key will also appear as soon as we implement the support of mutual TLS authentication for the normal Elastic Agents.

**Comment by nimarezainia** [comment]
> [@nimarezainia](https://github.com/nimarezainia) Could you help decide what is the priority on this one?

Done. I don't see this needing to be high priority tbh. Seems like the consensus is to have a separate helm for fleet server which is ok. @swiatekm what is the effort in addressing this? if it's not major perhaps we can prioritize higher for completeness sake.

**Comment by swiatekm** [comment]
Yeah, I was thinking of a very straightforward Chart without any modality. At best, we'd need to expose some [settings](https://github.com/elastic/fleet-server?tab=readme-ov-file#elasticsearch--kibana) as values in the Helm Chart. I'd honestly expect setting up the release process to be more work than the Chart itself. I think one sprint would be enough.

**Comment by MMollyy** [comment]
Seeing as the last post was May 2025, im gonna post asking if there is still progress being made on the helm chart? I've been at multiple locations of customers where they are deploying elastic and kibana on k8s and using helm charts, but then the fleet server doesn't have that capability yet.

**Comment by eedugon** [comment]
@MMollyy : As a side note, and FYI only, the "Elastic Stack" Helm chart provided by the ECK operator allows deploying Fleet Servers. More at https://www.elastic.co/docs/deploy-manage/deploy/cloud-on-k8s/managing-deployments-using-helm-chart

Not sure if that might be of any help, as maybe you don't want to rely on ECK operator for your Fleet Server deployment.
